initial onetrust implementation, need tests

rmi22186 opened the pull request:
Still need tests, but wanted to get this over to you for a directional PR for now. A few things:

- While I am PR-ing this into the template, I removed a lot of code that we didn't need to reduce the size of the file (there's no event, session, attribute, identity logging).

- this.process had to remain because in the core SDK always it always processes regardless of if it exists or not, which we can change. https://github.com/mParticle/mparticle-sdk-javascript/blob/master-v2/src/forwarders.js#L280

rmi22186 commented:
Also - OnConsentChanged will be called whenever certain other methods are called. Optanon.Close() calls a close on wahtever dialogue box may exist and then OnConsentChanged is fired.

rmi22186 commented:
@samdozor good for a review

samdozor commented on src/mParticleJSKitBoilerplate.js:
```diff
@@ -209,10 +54,10 @@ var UserAttributeHandler = require('../integration-builder/user-attribute-handle
-        this.setUserAttribute = setUserAttribute;
-        this.removeUserAttribute = removeUserAttribute;
         this.onUserIdentified = onUserIdentified;
+        this.process = function() {
```
is this needed?

rmi22186 replied:
this is needed because https://github.com/mParticle/mparticle-sdk-javascript/blob/master-v2/src/forwarders.js#L280

it doesn't do a check for `if forwarder[i].process`, it just runs it. events are still sent, they just disappear into this function now instead of erroring out. 

i can note to make a change in the core SDK to look for a `forwarder[i].process` and to only run it if it exists but for now we can keep this as is

samdozor replied:
cool 👍

samdozor commented on test/mParticle.js:
```diff
@@ -0,0 +1,6484 @@
+(function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;fo ...
```
is the full js sdk needed here or could you pull it in from `https://jssdkcdns.mparticle.com/js/v2/mparticle.js`?

rmi22186 replied:
this is needed because the link above has delete configureForwarder and delete addForwarder.  

in the future as a larger project i'm going to npm it all to simplify things and keep them all up to date.

samdozor commented on integration-builder/initialization.js:
```diff
@@ -1,33 +1,69 @@
+    },
+    createConsentEvents: function () {
+        var location = window.location.href,
+            consentState = mParticle.Consent.createConsentState(),
```
perhaps you should be querying for the user's current consent state and use that if not null, otherwise if purposes are set that are not configured via OneTrust then they'll be clobbered.

samdozor commented on integration-builder/initialization.js:
```diff
@@ -1,33 +1,69 @@
+            consent;
+
+        for (var key in consentMapping) {
+            var consentName = consentMapping[key];
```
maybe rename to consentPurpose?

rmi22186 commented:
1. generated build/OneTrustKit.js
2. updated readme
3. removed integration example
4. changed to oneTrustWrapper.js
5. moved all files to source

Saw the test failing and sorted out why also, so that change is in `initialization.js`, have to check to see if there is a user first

👀
